splink/expectation_step.py

```diff
@@ -88,12 +88,23 @@ def _sql_gen_gamma_prob_columns(params, settings, table_name="df_with_gamma"):
     select_cols = _add_left_right(select_cols, settings["unique_id_column_name"])
 
     for col in settings["comparison_columns"]:
-        col_name = col["col_name"]
-        if settings["retain_matching_columns"]:
-            select_cols = _add_left_right(select_cols, col_name)
-        if col["term_frequency_adjustments"]:
-            select_cols = _add_left_right(select_cols, col_name)
-        select_cols["gamma_" + col_name] = "gamma_" + col_name
+        if "col_name" in col:
```

Comment:
In these bits I'm having to fiddle around to make sure the order of the columns is sensible even when a custom_name is used